hnn_core/parallel_backends.py

```diff
@@ -57,32 +61,45 @@ def _gather_trial_data(sim_data, net, n_trials, postproc):
     """
     dpls = []
 
+    # Create array of equally sampled time points for simulating currents
+    cell_type_names = list(net.cell_types.keys())
+    cell_response = CellResponse(times=sim_data[0]['times'],
+                                 cell_type_names=cell_type_names)
+    net.cell_response = cell_response
+
     for idx in range(n_trials):
-        dpls.append(sim_data[idx][0])
-        sim_data_trial = sim_data[idx][1]
-        net.cell_response._spike_times.append(sim_data_trial[0])
-        net.cell_response._spike_gids.append(sim_data_trial[1])
+
+        # cell response
+        net.cell_response._spike_times.append(sim_data[idx]['spike_times'])
+        net.cell_response._spike_gids.append(sim_data[idx]['spike_gids'])
         net.cell_response.update_types(net.gid_ranges)
-        net.cell_response._vsoma.append(sim_data_trial[3])
-        net.cell_response._isoma.append(sim_data_trial[4])
+        net.cell_response._vsoma.append(sim_data[idx]['vsoma'])
+        net.cell_response._isoma.append(sim_data[idx]['isoma'])
+
+        # extracellular array
         for arr_name, arr in net.rec_arrays.items():
             # voltages is a n_trials x n_contacts x n_samples array
             net.rec_arrays[
-                arr_name]._data.append(sim_data_trial[5][arr_name])
+                arr_name]._data.append(sim_data[idx]['rec_data'][arr_name])
+            net.rec_arrays[arr_name]._times = sim_data[
+                idx]['rec_times'][arr_name]
 
-            net.rec_arrays[arr_name]._times = sim_data_trial[6][arr_name]
+        # dipole
+        dpl = Dipole(times=sim_data[idx]['times'],
+                     data=sim_data[idx]['dpl_data'])
```

Comment:
rearranged the function so that dipole, cell response and extracellular array are in clearly demarcated blocks. Also I find it neater to use the following pattern:

```py
dpl = Dipole(...)
dpl.blah1()
dpl.blah2()
dpls.append(dpl)
```

rather than:

```py
dpls.append(Dipole(...))
dpls[-1].blah1()
dpls[-1].blah2()
```

maybe just a personal preference though

Reply:
+1 to avoid those confusing `[-1]`s!